jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/controllers/DslWeightedSwitchController.java

```diff
@@ -0,0 +1,107 @@
+package us.abstracta.jmeter.javadsl.core.controllers;
+
+import com.blazemeter.jmeter.control.WeightedSwitchController;
+import com.blazemeter.jmeter.control.WeightedSwitchControllerGui;
+import org.apache.jmeter.gui.util.PowerTableModel;
+import org.apache.jmeter.testelement.TestElement;
+import org.apache.jorphan.collections.HashTree;
+import us.abstracta.jmeter.javadsl.core.BuildTreeContext;
+import us.abstracta.jmeter.javadsl.core.testelements.BaseTestElement;
+import us.abstracta.jmeter.javadsl.core.threadgroups.BaseThreadGroup.ThreadGroupChild;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * DSL wrapper for {@link WeightedSwitchController}
+ */
+public class DslWeightedSwitchController extends BaseController {
+
+  private final PowerTableModel model;
+  private final WeightedSwitchController wsc;
+
+  public DslWeightedSwitchController() {
+    super("Weighted Switch Controller",
+        WeightedSwitchControllerGui.class,
+        new ArrayList<>()
+    );
+
+    model = new PowerTableModel(
+        new String[]{"Name", "Weight", "Enabled"},
+        new Class[]{String.class, String.class, String.class}
+    );
+
+    wsc = new WeightedSwitchController();
+  }
+
+  /**
+   * Internal method to add TestElement to {@link WeightedSwitchController} model
+   *
+   * Checks that total sum of weights will not exceed 100 considering the new element
+   *
+   * @param weight  weight of new TestElement in the model
+   * @param element the new TestElement
+   * @param enabled flag to set TestElement enabled or not
+   *
+   * @throws IllegalArgumentException throws exception if total sum of weights exceed 100
+   */
+  private void addToModel(Long weight, BaseTestElement element, Boolean enabled) throws IllegalArgumentException {
+
+    List<Long> weights = this.model.getData().getColumnAsObjectArray("Weight")
+        .stream()
+        .map(x -> Long.parseLong(x.toString()))
+        .collect(Collectors.toList());
+
+    if (weights.stream().reduce(0L, Long::sum) + weight <= 100) {
+      this.model.addRow(new String[]{element.getName(), weight.toString(), enabled.toString()});
```

Comment:
I wonder if we might implement something but without using getName... something with buildTreeUnder and reviewing test elements in hashtree. I would like if possible to rollback the change of adding getName and getGuiClass if possible to reduce api methods and simplify it.